test/signalrclienttests/url_builder_tests.cpp

```diff
@@ -101,6 +101,118 @@ TEST(url_builder_connect, url_correct_if_connection_data_not_empty)
+        url_builder::build_reconnect(web::uri{ _XPLATSTR("http://fake/signalr/") }, transport_type::long_polling,
+        _XPLATSTR("connection token"), _XPLATSTR("[{\"Name\":\"ChatHub\"}]"), _XPLATSTR(""), _XPLATSTR(""), _XPLATSTR("")));
+
+    ASSERT_EQ(
+        web::uri(_XPLATSTR("ws://fake/signalr/reconnect?transport=webSockets&clientProtocol=1.4&connectionToken=connection%20token&connectionData=%5B%7B%22Name%22:%22ChatHub%22%7D%5D")),
+        url_builder::build_reconnect(web::uri{ _XPLATSTR("http://fake/signalr/") }, transport_type::websockets,
+        _XPLATSTR("connection token"), _XPLATSTR("[{\"Name\":\"ChatHub\"}]"), _XPLATSTR(""), _XPLATSTR(""), _XPLATSTR("")));
+}
+
+TEST(url_builder_reconnect, url_correct_if_last_message_id_not_empty)
+{
+    ASSERT_EQ(
+        web::uri(_XPLATSTR("http://fake/signalr/reconnect?transport=longPolling&clientProtocol=1.4&connectionToken=connection%20token&connectionData=%5B%7B%22Name%22:%22ChatHub%22%7D%5D&messageId=L45T%20M355463_1D")),
+        url_builder::build_reconnect(web::uri{ _XPLATSTR("http://fake/signalr/") }, transport_type::long_polling,
+        _XPLATSTR("connection token"), _XPLATSTR("[{\"Name\":\"ChatHub\"}]"), _XPLATSTR("L45T M355463_1D"), _XPLATSTR(""), _XPLATSTR("")));
+
+    ASSERT_EQ(
+        web::uri(_XPLATSTR("ws://fake/signalr/reconnect?transport=webSockets&clientProtocol=1.4&connectionToken=connection%20token&connectionData=%5B%7B%22Name%22:%22ChatHub%22%7D%5D&messageId=L45T%20M355463_1D")),
+        url_builder::build_reconnect(web::uri{ _XPLATSTR("http://fake/signalr/") }, transport_type::websockets,
+        _XPLATSTR("connection token"), _XPLATSTR("[{\"Name\":\"ChatHub\"}]"), _XPLATSTR("L45T M355463_1D"), _XPLATSTR(""), _XPLATSTR("")));
+}
+
+TEST(url_builder_reconnect, url_correct_if_groups_token_not_empty)
+{
+    ASSERT_EQ(
+        web::uri(_XPLATSTR("http://fake/signalr/reconnect?transport=longPolling&clientProtocol=1.4&connectionToken=connection%20token&connectionData=%5B%7B%22Name%22:%22ChatHub%22%7D%5D&groupsToken=G%207")),
+        url_builder::build_reconnect(web::uri{ _XPLATSTR("http://fake/signalr/") }, transport_type::long_polling,
+        _XPLATSTR("connection token"), _XPLATSTR("[{\"Name\":\"ChatHub\"}]"), _XPLATSTR(""), _XPLATSTR("G 7"), _XPLATSTR("")));
+
+    ASSERT_EQ(
+        web::uri(_XPLATSTR("ws://fake/signalr/reconnect?transport=webSockets&clientProtocol=1.4&connectionToken=connection%20token&connectionData=%5B%7B%22Name%22:%22ChatHub%22%7D%5D&groupsToken=G%207")),
+        url_builder::build_reconnect(web::uri{ _XPLATSTR("http://fake/signalr/") }, transport_type::websockets,
+        _XPLATSTR("connection token"), _XPLATSTR("[{\"Name\":\"ChatHub\"}]"), _XPLATSTR(""), _XPLATSTR("G 7"), _XPLATSTR("")));
+}
+
+TEST(url_builder_reconnect, query_string_added_after_message_id_and_groups_token)
+{
+    auto x = url_builder::build_reconnect(web::uri{ _XPLATSTR("http://fake/signalr/") }, transport_type::long_polling,

```

Comment:
The whole line is not needed. I added it when debugging (could not get the expression to evaluate in the watch window) and forgot to remove. Will remove.